hubspot3/tickets.py

```diff
@@ -78,3 +90,171 @@ def get_all(self, limit: int = -1, **options) -> list:
             offset = batch["offset"]
 
         return output if not limited else output[:limit]
+
+    default_batch_properties = [
+        "subject",
+    ]
+
+    def _get_batch(self, ids, extra_properties: Union[list, str] = None,
+                   with_history: bool = False):
+        """given a batch of vids, get more of their info"""
+        # default properties to fetch
+        properties = set(self.default_batch_properties)
+
+        # append extras if they exist
+        if extra_properties:
+            if isinstance(extra_properties, list):
+                properties.update(extra_properties)
+            if isinstance(extra_properties, str):
+                properties.add(extra_properties)
+        if with_history:
+            property_name = 'propertiesWithHistory'
+        else:
+            property_name = 'properties'
+        if len(properties) > 0:
+            params = {property_name: list(properties)}
+        else:
+            params = {}
+        batch = self._call(
+            "objects/tickets/batch-read",
+            method="POST",
+            doseq=True,
+            params=params,
+            data={'ids': ids}
+        )
+        # Returning a dict with IDs as keys
+        return batch
+
+    def get_batch(self, ids, extra_properties: Union[list, str] = None):
+        """given a batch of ticket, get more of their info"""
+        batch = self._get_batch(ids, extra_properties=extra_properties, with_history=False)
+        return [prettify(batch[ticket], id_key="objectId") for ticket in batch]
+
+    def get_batch_with_history(self, ids, extra_properties: Union[list, str] = None):
+        """given a batch of ticket, get more of their info with history"""
+        batch = self._get_batch(ids, extra_properties=extra_properties, with_history=True)
+        return batch
+
+    def _get_recent(
+        self,
+        recency_type: str,
+        limit: int = -1,
+        vid_offset: int = 0,
+        time_offset: int = 0,
+        max_time_diff_ms: int = 60000,
+        # Correct default value waiting for
+        # https://community.hubspot.com/t5/APIs-Integrations/Time-difference-between-ticket-s-quot-get-log-of-changes-quot/m-p/297909#M27982
+        **options
+    ):
+
+        finished = False
+        query_limit = 100  # max according to the docs
+        limited = limit > 0
+        if limited and limit < query_limit:
+            query_limit = limit
+        total_tickets = 0
+        while not finished:
+            params = {"count": query_limit}
+            if vid_offset:
+                params["objectId"] = vid_offset
+            if time_offset:
+                params["timestamp"] = int(time_offset)
+            params['changeType'] = recency_type
+            batch = self._call(
+                "change-log/tickets",
+                method="GET",
+                params=params,
+                doseq=True,
+                **options
+            )
+
+            total_tickets += len(batch)
```

Comment:
You are using `len(batch)` three lines in a row. As it's not being changed, you could extract it to a variable.